source/hp/refinement.cc

```diff
@@ -45,6 +45,10 @@ namespace hp
     void
     full_p_adaptivity(const dealii::DoFHandler<dim, spacedim> &dof_handler)
     {
+      if (dof_handler.get_fe_collection().size() == 0)
+        // nothing to do
+        return;
```

Comment:
Here and below, wouldn't the condition `==0` imply that the `DoFHandler` has never been initialized via `distribute_dofs()`? I'd consider that an error in this function and just assert so, rather than exiting without doing anything.

Reply:
> Here and below, wouldn't the condition ==0 imply that the DoFHandler has never been initialized via distribute_dofs()?

Yes.

> I'd consider that an error in this function and just assert so, rather than exiting without doing anything.

I'm okay with your proposed assertions for most of the `hp::Refinement` functions.

However, my plan is to attach the `predict_error()` and `limit_p_level_difference()` functions to the new `post_p4est_refinement` signal. Whenever calling the `refine_global` function to create an initial mesh, these functions will ultimately be called and at that stage dofs have not been enumerated yet. This was the original motivation of this PR :-)

I don't know how to deal with this other than exiting without doing anything. Do you have a different idea?

Reply:
> I don't know how to deal with this other than exiting without doing anything. Do you have a different idea?

Hmm, users could simply connect these functions to the signal after generating the initial mesh. But this feels more like a hotfix than a solution.